src/Esprima/ParserExtensions.cs

```diff
@@ -1,640 +1,55 @@
-﻿using System.Runtime.CompilerServices;
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
 
 namespace Esprima;
 
 public static class ParserExtensions
 {
-    private static readonly string[] charToString = new string[256];
+    private static readonly string[] s_charToString = new string[256];
 
     static ParserExtensions()
     {
-        for (var i = 0; i < charToString.Length; ++i)
+        for (var i = 0; i < s_charToString.Length; ++i)
         {
             var c = (char) i;
-            charToString[i] = c.ToString();
+            s_charToString[i] = c.ToString();
         }
     }
 
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static string? TryGetInternedString(ReadOnlySpan<char> source)
+    {
+        return
```

Comment:
I think this must be quite costly to fall back and do all the method calls which have switches for possible same string lengths too? Ideally we would have separate method that has all the dictionary words we want to support?